sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3548,8 +3607,11 @@ pi_result piKernelCreate(pi_program Program, const char *KernelName,
     return PI_ERROR_UNKNOWN;
   }
 
-  // Update the refcount of the program to show its use by this kernel.
+  // Update the refcount of the program and context to show it's used by this
+  // kernel.
   PI_CALL(piProgramRetain(Program));
+  if (IndirectAccessSupportEnabled)
```

Comment:
I think it's a good idea to always retain context for the duration of the kernel. If it's not easy to do now, then disregard this comment